### DISCUSSION TITLE
Missing OpenJDK11

### DISCUSSION THREAD

**Discussion by drei97** [unoplatform/uno#17666] [Q&A]
Hi,

I am new to UNO and want to try out if it fits my needs and expectations. 

My system is a Linux, OpenSuSE Tumbleweed and the preferred IDE is Jetbrains Rider.

After installing the templates I started uno-check and there is still a missing dependency. The message is

────────────────────────────────────────────────────────────────────────────
¤ Recommendation: Install OpenJDK11
OpenJDK 11 is missing, follow the installation instructions here: https://learn.microsoft.com/en-us/java/openjdk/install#install-on-ubuntu
────────────────────────────────────────────────────────────────────────────

On my system is installed:

```
  # rpm -qa | grep jdk
java-21-openjdk-devel-21.0.4.0~6-1.1.x86_64
java-21-openjdk-21.0.4.0~6-1.1.x86_64
java-11-openjdk-headless-11.0.24.0~7-1.1.x86_64
java-11-openjdk-11.0.24.0~7-1.1.x86_64
java-21-openjdk-headless-21.0.4.0~6-1.1.x86_64
java-11-openjdk-devel-11.0.24.0~7-1.1.x86_64
  #
```

What is missing, or what is uno-check doing in detail to check if OpenJDK11 is available? Maybe I then will be able to fix.

Joachim

**Comment by MartinZikmund** [comment]
@drei97 it is quite possible our checkup is incorrect for OpenSuSE. Here is the code that verifies OpenJDK - https://github.com/unoplatform/uno.check/blob/main/UnoCheck/Checkups/OpenJdkCheckup.cs - can you check what is the difference on your system?

**Reply by drei97** [reply]
OK, the constructor in 

```
var xamSdkInfo = new AndroidSdkInfo((traceLevel, msg) => Util.Log(msg), null, null, null);
```

of methode OpenJdkCheckup.Examine throws an exception. 

But this is not due to a missing OpenJDK but due to missing AndroidSDK which I do not need.

Here the full log from Debug mode

```
› OpenJDK 11.0 Checkup...
ValidateAndroidSdkLocation: ``, result=False
ValidateAndroidSdkLocation: ``, result=False
ValidateAndroidSdkLocation: `/home/joachim/Dokumente/Library/Android/sdk`, result=False
ValidateJavaSdkLocation: ``, result=False
ValidateJavaSdkLocation: ``, result=False
ValidateJavaSdkLocation: `/usr/lib64/jvm/java-21-openjdk-21`, result=True
ValidateAndroidNdkLocation: ``, result=False
ValidateAndroidNdkLocation: ``, result=False
System.InvalidOperationException: Could not determine Android SDK location. Please provide `androidSdkPath`.
  at Xamarin.Android.Tools.AndroidSdkInfo..ctor(Action<TraceLevel, string> logger, string androidSdkPath, string androidNdkPath, string   
     javaSdkPath)                                                                                                                         
  at Task<DiagnosticResult> DotNetCheck.Checkups.OpenJdkCheckup.Examine(SharedState history) in                                           
     /home/joachim/nas/08_source/03_Download/uno.check/UnoCheck/Checkups/OpenJdkCheckup.cs:63                                             


```

So the error message of uno-check in Release mode is misleading!

**Comment by drei97** [comment]
BTW.: To get uno-check compiled and running I had to change some hard coded window paths like "D:\" which are not working under Linux. I am not sure if it is intended that people compile uno by themselves. But maybe it make sense to generalize the paths.

**Reply by jeromelaban** [reply]
Thanks for the update. The path you're mentioning is in `launchSettings.json` and it needs to be adjusted per dev machine at this time. It's there only to provide an example on what the parameters can be to test the project.

**Reply by drei97** [reply]
Since the requested file is within the solution, a relalive path could do the job?

**Reply by jeromelaban** [reply]
I don't recall the details, but for windows because of elevation, a relative path might not have been enough in the past.
